addons/knowledge/views/knowledge_templates_common.xml

```diff
@@ -66,7 +66,20 @@
     </template>
 
     <template id="articles_template_name_readonly">
-        <div class="o_article_name flex-grow-1 text-truncate o_cursor_pointer">
+        <t t-if="not portal_readonly_mode">
```

Comment:
Strange to test "portal_readonly_mode" in a template that is called "articles_template_name_readonly"...

Isn't this one *always* readonly? The name would make you think so ^^

Reply:
Yes I know that.

My point was that this template in **only called in readonly** currently.
See: https://github.com/odoo/odoo/blob/master/addons/knowledge/views/knowledge_templates_common.xml#L34

It's strange to copy/paste part of the code from "articles_template_name_editable".
Either it's editable, either it's not, but here it seems we have a weird mix.

Reply:
Ah, I see. The original issue was that if we leave `<a [...] t-attf-href="/knowledge/article/{{article.id}}"` for internal users, when clicking on a read-only article it will leave the page and refresh the whole view, which we really don't want. I can't really think of a more simple and elegant solution.

Reply:
We could have a single template instead and add some conditioning based on the readonly state if that is not too complicated?

Reply:
Could we maybe move this to another task? There are a lot of other fixes/improvements to do with templates

Reply:
Well either we do it cleanly here or we don't do any change and address this later.
But copy/pasting half a template to be cleaned later means we probably will never do it ^^